gateway/ln-gateway/src/lib.rs

```diff
@@ -1018,13 +1018,7 @@ impl Gateway {
 
         // The gateway deterministically assigns a channel id (u64) to each federation
         // connected.
-        let mut next_scid = self.federation_manager.next_scid.lock().await;
-        let mint_channel_id = *next_scid;
-        *next_scid = next_scid
-            .checked_add(1)
-            .ok_or(GatewayError::GatewayConfigurationError(
-                "Too many connected federations".to_string(),
-            ))?;
+        let mint_channel_id = self.federation_manager.pop_next_scid().await?;
```

Comment:
Note for reviewers: I believe in the past `next_scid` was used to ensure synchronization when adding federations. However, above this we already claim `Gateway.client_joining_lock`. Since that lock is claimed before locking on `next_scid`, it should always be more restrictive, so any locking guarantees provided by `next_scid` shouldn't be needed.